### PR TITLE
Multiple updates to setup_zoc_mn.sh

### PR DESCRIPTION
**Pull request by deadthings**
- Reorganized script to reflect execution logic
- Added sudo install
- Added dnsutils package for dig command
- Further improvements to printed text
- Added clarity to pre-existing privkey instructions
- Revised method for adding mnchecker to cron
- Reduced maxconnections from 25 to 16 to help with resource allocation on small VPS servers
- Changed sequence to install sentinel earlier in the process
- Relevant info printed at end of script all in one place

### CODE REVIEW

File: setup_zoc_mn.sh [hunk 1]
```diff
@@ -26,24 +86,25 @@ configQuestions()
...
+			[Yy]* ) echo "Please type in your private key below:"; askforprivatekey;break;;
 			[Nn]* ) genkey;break;;
 			* ) echo "Please answer yes or no.";;
 		esac
 	done
 
 	echo "**************************************************************"
 	while true; do
-		read -p "Would you like a shell scipt to start the node?  [Y/N]" yn
+		read -p "Would you like a shell scipt to start the node?  [Y/N] " yn
```
Comment (camenlx) [inline setup_zoc_mn.sh:107]: scipt -> script

File: setup_zoc_mn.sh [hunk 2]
```diff
@@ -26,24 +86,25 @@ configQuestions()
...
+		mkdir ~/.zeroonecore 
 	fi
 	
 	echo "rpcuser=${rpcuser}" >> .zeroonecore/zeroone.conf
 	echo "rpcpassword=${rpcpassword}" >> .zeroonecore/zeroone.conf
 	
+	echo "**************************************************************"
 	while true; do
-		read -p "Would you like to provide a private key (if you already have one)? ${privkey} [Y/N]" yn
+		read -p "Would you like to provide a private key (if you already have one)? ${privkey} [Y/N] " yn
```
Comment (camenlx) [inline setup_zoc_mn.sh:97]: personally I do prefer call this parameter a 'genkey'

File: setup_zoc_mn.sh [hunk 3]
```diff
@@ -4,6 +4,66 @@ echo "                     01coin Masternode Setup Shell Script"
...
+{
+	echo "*******************************************************************************"
+	echo "                           Installing Requirements"
+	echo "*******************************************************************************"
+
+	apt install -y sudo
+	sudo apt install -y software-properties-common
+	sudo add-apt-repository -y ppa:bitcoin/bitcoin
+	sudo apt update 
+	sudo apt upgrade -y
```
Comment (camenlx) [inline setup_zoc_mn.sh:17]: I would not propose an 'upgrade' ...

File: setup_zoc_mn.sh [hunk 4]
```diff
@@ -4,6 +4,66 @@ echo "                     01coin Masternode Setup Shell Script"
...
+	
+	mkdir ~/.zeroonecore
+}
+
+compile()
+{
+#checks ram and makes swap if nessary
+	ram="$(awk '/MemTotal/ {print $2}' /proc/meminfo)"
+	minram="1048576"
+	if [ "$ram" -le "$minram" ]
```
Comment (camenlx) [inline setup_zoc_mn.sh:46]: I would ask user permission to create a swapfile

File: setup_zoc_mn.sh [hunk 5]
```diff
@@ -62,37 +123,63 @@ configQuestions()
...
+genkey()
+{
+		zeroone/zerooned -daemon
+		echo "Please wait while the 01coin daemon generates a new private key..."
+		sleep 10
+		while ! zeroone/zeroone-cli getinfo; do
+			sleep 10
+		done
+		privkey=$(zeroone/zeroone-cli masternode genkey)
+		killall -9 zerooned
```
Comment (camenlx) [inline setup_zoc_mn.sh:169]: why use killall and not normal cli stop?
killall leaves .lock and will kill also testnet daemon

File: setup_zoc_mn.sh [hunk 6]
```diff
@@ -62,37 +123,63 @@ configQuestions()
...
+
+setup_manager()
+{
+	killall zerooned
+	wget https://raw.githubusercontent.com/zocteam/zoc-tools/master/mnchecker
+	chmod 777 mnchecker
+	echo "rpcport=10101" >> .zeroonecore/zeroone.conf
+
+	crontab -l > mncheckercron
+	echo "*/10 * * * * /root/mnchecker --currency-bin-cli=/root/zeroone/zeroone-cli --currency-bin-daemon=/root/zeroone/zerooned --currency-datadir=.zeroonecore" >> mncheckercron
```
Comment (camenlx) [inline setup_zoc_mn.sh:180]: will manager only works as root?

**Review by camenlx — COMMENTED** [review]
Please Evy also consider my review comments.

File: setup_zoc_mn.sh [hunk 7]
```diff
@@ -62,37 +123,63 @@ configQuestions()
...
-	echo "*******************************************************************************"
-	echo "                           Installing Requirements"
-	echo "*******************************************************************************"
+#Masternode Priv Key
+	read privkey
+	while true; do
+		read -p "Is this the correct private key? ${privkey} [Y/N] " yn
+		case $yn in
+			[Yy]* ) break;;
+			[Nn]* ) echo "Please type in your VPS IP address below:"; read privkey;;
```
Comment (camenlx) [inline setup_zoc_mn.sh:134]: "your VPS IP address below:" ->  "your generated private key below:"

**Comment by Evydder** [comment]
Applied Camenlx comments in another commit

**Pull request merged**